File: qb-engineer-server/qb-engineer.core/Models/SalesOrderShipmentModel.cs
namespace QBEngineer.Core.Models;

public record SalesOrderShipmentModel(
    int Id,
    string ShipmentNumber,
    string Status,
    string? Carrier,
    string? TrackingNumber,
    DateTimeOffset? ShippedDate,
    DateTimeOffset? DeliveredDate,
    decimal ShippingCost,
    decimal? Weight,
    string? Notes,
    List<SalesOrderShipmentLineModel> Lines,
    List<SalesOrderShipmentPackageModel> Packages);

File: qb-engineer-ui/src/app/features/sales-orders/models/sales-order-shipment.model.ts
export interface SalesOrderShipment {
  id: number;
  shipmentNumber: string;
  status: string;
  carrier: string | null;
  trackingNumber: string | null;
  shippedDate: Date | null;
  deliveredDate: Date | null;
  shippingCost: number;
  weight: number | null;
  notes: string | null;
  lines: SalesOrderShipmentLine[];
  packages: SalesOrderShipmentPackage[];
}

export interface SalesOrderShipmentLine {
  id: number;
  partId: number | null;
  partNumber: string | null;
  quantity: number;
  notes: string | null;
  salesOrderLineId: number | null;
}

export interface SalesOrderShipmentPackage {
  id: number;
  trackingNumber: string | null;
  carrier: string | null;
  weight: number | null;
  length: number | null;
  width: number | null;
  height: number | null;
  status: string | null;
}
